thingif/src/test/java/com/kii/thingif/thingifapi/ListTriggeredServerCodeResultTest.java

```diff
@@ -0,0 +1,189 @@
+package com.kii.thingif.thingifapi;
+
+import android.content.Context;
+import android.util.Pair;
+
+import com.kii.thingif.ServerError;
+import com.kii.thingif.StandaloneThing;
+import com.kii.thingif.Target;
+import com.kii.thingif.ThingIFAPI;
+import com.kii.thingif.ThingIFAPITestBase;
+import com.kii.thingif.TypedID;
+import com.kii.thingif.exception.NotFoundException;
+import com.kii.thingif.thingifapi.utils.ThingIFAPIUtils;
+import com.kii.thingif.trigger.TriggeredServerCodeResult;
+import com.kii.thingif.trigger.TriggeredServerCodeResultFactory;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import com.squareup.okhttp.mockwebserver.RecordedRequest;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+@RunWith(RobolectricTestRunner.class)
+public class ListTriggeredServerCodeResultTest extends ThingIFAPITestBase{
+
+    private Context context;
+
+    @Before
+    public void before() throws Exception {
+        context = RuntimeEnvironment.application.getApplicationContext();
+        this.server = new MockWebServer();
+        this.server.start();
+    }
+
+    @After
+    public void after() throws Exception {
+        this.server.shutdown();
+    }
+
+    @Test
+    public void listTriggeredServerCodeResultsTest() throws Exception {
+        String triggerID = UUID.randomUUID().toString();
+        TypedID thingID = new TypedID(TypedID.Types.THING, "th.1234567890");
+        String accessToken = "thing-access-token-1234";
+        Target target = new StandaloneThing(thingID.getID(), "vendor-thing-id", accessToken);
+        String paginationKey = "pagination-12345-key";
+
+        ThingIFAPI api = this.createDefaultThingIFAPI(this.context, APP_ID, APP_KEY);
+        ThingIFAPIUtils.setTarget(api, target);
+
+        TriggeredServerCodeResult serverCodeResult1 =
```

Comment:
This is cosmetic so you do not need to adopt my suggestion:)

I think following code is more simple.

```java
TriggeredServerCodeResult[] firstResults[] = new TriggeredServerCodeResult[] {
    TriggeredServerCodeResultFactory.create(true, "1234", System.currentTimeMillis(), "func1", null),
    TriggeredServerCodeResultFactory.create(true, "12.34", System.currentTimeMillis() 1000, "func2", null),
    TriggeredServerCodeResultFactory.create(true, "true", System.currentTimeMillis() 2000, "func3", null),
    TriggeredServerCodeResultFactory.create(true, "\"abcd\"", System.currentTimeMillis() 3000, "func4", null)
};
TriggeredServerCodeResult[] secondResults[] = new TriggeredServerCodeResult[] {
    TriggeredServerCodeResultFactory.create(true, "{\"field\":\"abcd\"}", System.currentTimeMillis() 4000, "func5", null),
    TriggeredServerCodeResultFactory.create(true, "[1, \"2\", 3]", System.currentTimeMillis() 5000, "func6", null),
    TriggeredServerCodeResultFactory.create(false, null, System.currentTimeMillis() 6000, "func7", new ServerError("Error found", "RUNTIME_ERROR", "faital error")),
};
addMockResponseForListTriggeredServerCodeResults(200, firstResults, paginationKey);
addMockResponseForListTriggeredServerCodeResults(200, secondResults, null);

Pair<List<TriggeredServerCodeResult>, String> result1 = api.listTriggeredServerCodeResults(triggerID, 4, null);
Assert.assertEquals(paginationKey, result1.second);
Assert.assertEquals(firstResults.length, result1.first.size());
for (int i = 0 ; i < firstResults.length; ++i) {
    assertSameTriggeredServerCodeResults(firstResults[i], result1.first.get(i));
}

Pair<List<TriggeredServerCodeResult>, String> result2 = api.listTriggeredServerCodeResults(triggerID, 4, result1.second);
Assert.assertNull(result2.second);
Assert.assertEquals(secondResults.length, result2.first.size());
for (int i = 0 ; i < secondResults.length; ++i) {
    assertSameTriggeredServerCodeResults(secondResults[i], result2.first.get(i));
}
```